sup3r/pipeline/forward_pass.py

```diff
@@ -736,8 +737,26 @@ def __init__(self, file_paths, model_kwargs, fwp_chunk_shape,
                                             self.spatial_pad,
                                             self.temporal_pad)
 
+        msg = ('The average memory of a forward pass output chunk is '
+               f'{self.output_chunk_mem / 1e9:.2f} GB. Exceeding 1.5 GB could'
+               ' result in constant model output.')
+        if self.output_chunk_mem / 1e9 > 1.5:
+            logger.warning(msg)
+            warnings.warn(msg)
```

Comment:
I have mixed feelings about this, do we know what's actually causing this? Is this a documented issue? I can't find anything online. 

Most importantly, the size of the output chunk at 1.5 GB is likely not the cause of this problem. It's almost certainly a memory overflow mid-network where the tensor is much larger. If we can confirm that tensorflow basically does a silent OOM error, we should just include a psutil check in the model.generate() method between layers.

Reply:
I haven't been able to find anything either. I guess you're right that its probably not dependent on output size but some intermediate size. I'd prefer we do a check before rather than waiting to see if the output is bad. It's very clear that it's related to memory though since just increasing the padding passed a threshold gives constants all of a sudden.

Reply:
But if you do the check before the first forward pass you're assuming A) this threshold you've set is going to be applicable to any model architecture (bad assumption) and B) the user is going to be parsing the log files for warning statements (unlikely if you're running over night). 

If you raise an exception in response to a bad chunk output you will 1) save a lot of compute time because the job won't continue to run, 2) save the human time of having to monitor a huge log file for really bad messages, and 3) have a check that works for any model architecture.

Reply:
Yeah I definitely agree on the first part. Im hoping there is a model agnostic pre check we could do. If we do a check on output then the only compute were avoiding is the h5 write.

Reply:
I think there's a misunderstanding - we should check every chunk output, so the exception would be raise after the first forward pass chunk is complete and all other chunks would not be run.

Reply:
Yeah this makes sense. I was thinking that not every chunk was guaranteed to fail but nearly all will, except possibly the ones on the edges.